geomstats/geometry/discrete_curves.py

```diff
@@ -404,11 +607,13 @@ def diffeomorphism(self, base_point):
             SRV representation.
         """
         ndim = self._point_ndim
-        base_point_with_origin = insert_zeros(base_point, axis=-ndim)
-
+        base_point_with_origin = self.space.insert_origin(base_point)
+        if self.space.closed:
+            base_point_with_origin = insert_starting_point_at_end(
+                base_point_with_origin
+            )
```

Comment:
If this increases the number of sampling points of one, you may need to change some normalizations constants that depend on this number of sampling points.